---
title: "A Gentle Introduction to Spatial Transcriptomics Analysis"
author: "Jessica Gillespie"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Welcome!
Future home of a simple, (hopefully) easy to understand spatial transcriptomics analysis pipeline for those who have never before done ST analysis.

This started as part of my masters thesis but has been continued as a way to provide some guidance to all of those trying to teach themselves a new NGS analysis techniques. The first version of this protocol will soon be published (pending link).

## Who is this NOT for
This protocol is not for anyone with extensive ST analysis experience. We will not be discussing intricate details or highly technical aspects beyond general notes to new users or "tips and tricks".

Conversely, this protocol is also not for someone completely new to next-generation sequencing (NGS), bioinformatics (BMI) analysis, or R. We will not be covering in detail datatypes, file handling, basic biological concepts involved with ST samples, or ST concepts prior to completed sequencing. There are places to find such information and repeating them here would be redundant and beyond the scope of this project. Instead, two good resources are [W3Schools](https://www.w3schools.com/r/default.asp) which will teach you the very basics of R and [DataQuest](https://www.dataquest.io/blog/learn-r-for-data-science/) which has more basics and also some simple projects where you can practice coding.

## So who is this actually for then?
The idea of this protocol is to run someone through their "first time" doing spatial transcriptomics analysis. Maybe you want to see if this is a new experiment type you would like to pursue in your lab. Maybe you just want to dabble in a new method to expand your BMI repertoire. If you want to take a simple data set and get a small hands-on experience with ST analysis, this is for you.

## Want to contribute?
The world of ST analysis is ever changing and constantly expanding. There is no guarantee this page or project will contain the most up to date information. This project is also currently managed by a single person in their spare time. If you would like to make updates, corrections, report problems, or suggest improvements, please feel free to do so using the issue tracker or discussion boards here on github or (if you are comfortable with git and coding) contributing directly to the repository.

There are several plans for the expansion of this protocol and they will be implemented as time allows. However, I hope you find this project helpful and informative in its current state.

## About this actual site
These pages are all written in R using a package called [Bookdown](https://bookdown.org/). This package allows you to write R scripts that can then be converted into html files. These files can then be uploaded to GitHub and published via GitPages. The R files are available for download on GitHub if you want a "behind the scenes" view of these pages or you want to run them without copy/pasting code from these pages.





At the bottom of every chapter, you will see output from `sessionInfo()`. This is a command in R that prints a list of current R environment infomration as well as packages that were loaded at the time of creating this page. While it is not necessary if you are creating simple webpages like this one, it is provided as part of the protocol as it can be useful in troubleshooting.

``` {r session}
sessionInfo()
```

<!--chapter:end:index.Rmd-->

